db/src/logger.rs

```diff
@@ -172,24 +179,72 @@ impl Logger {
         }
 
         let temp_path = inner.config.compaction_location()?;
-        let final_path = inner.config.db_location()?;
+        let final_path = inner.config.db_location_v2()?;
 
         let mut file = Self::open_file(&inner.config, &temp_path)?;
-        let data = Self::log_entry(0, data);
-        file.write_all(&data)?;
 
+        // write compaction count for future IPC reasons
+        let mut log_meta = inner
+            .log_metadata
+            .ok_or(DbError::Unexpected("log meta missing -- no_io == false"))?;
+        log_meta.compaction_count += 1;
+        let metadata_bytes = log_meta.to_bytes();
+        file.write_all(&metadata_bytes)?;
+
+        // write compacted data to a temporary file
+        let compacted_data = Self::log_entry(0, data);
+        file.write_all(&compacted_data)?;
+
+        // atomically make this the new log
         fs::rename(temp_path, final_path)?;
         inner.file = Some(file);
+        inner.log_metadata = Some(log_meta);
+
+        Ok(())
+    }
+
+    fn handle_migration(config: &Config) -> DbResult<()> {
+        let v1 = config.db_location_v1()?;
+        let v2 = config.db_location_v2()?;
+        let v2_temp = PathBuf::from(format!("{}.migration", v2.to_string_lossy()));
+
+        if !v1.exists() {
+            return Ok(());
+        }
+
+        if v2_temp.exists() {
+            fs::remove_file(&v2_temp)?;
+        }
+
+        if v2.exists() {
+            return Err(DbError::Unexpected("Found v1 log alongside v2 log!"));
```

Comment:
This is a problem:
- presently we don't delete the old log
- even if we did, I'm not sure we could have atomic certainty that this case wouldn't be hit in the event of interrupts
- to have this be fully consistent, we would probably have to just log in this event and use the new log by default